scripts/migration.sh

```diff
@@ -33,6 +33,16 @@ fi
 printf "Generating migration boilerplate for patch level $NEW_LEVEL..."
 
 echo "SET NAMES utf8mb4 COLLATE utf8mb4_bin;\n" > "$FWD_SCHEMA"
+echo "DROP PROCEDURE IF EXISTS auto_check_migration_sanity_$NEW_LEVEL;\n" >> "$FWD_SCHEMA"
+echo "CREATE PROCEDURE auto_check_migration_sanity_$NEW_LEVEL()" >> "$FWD_SCHEMA"
+echo "BEGIN" >> "$FWD_SCHEMA"
+echo "  SELECT @currentPatchLevel := value FROM dbMetadata WHERE name = 'schema-patch-level';" >> "$FWD_SCHEMA"
+echo "  IF @currentPatchLevel != '$PREV_LEVEL' THEN" >> "$FWD_SCHEMA"
+echo "    SIGNAL SQLSTATE '45000' SET MYSQL_ERRNO = 1643, MESSAGE_TEXT = 'Missing migration detected';" >> "$FWD_SCHEMA"
+echo "  END IF;" >> "$FWD_SCHEMA"
+echo "END;\n" >> "$FWD_SCHEMA"
+echo "CALL auto_check_migration_sanity_$NEW_LEVEL;\n" >> "$FWD_SCHEMA"
+echo "DROP PROCEDURE auto_check_migration_sanity_$NEW_LEVEL;\n" >> "$FWD_SCHEMA"
```

Comment:
This looks fine, but I worry about about the readability of patch files if they grow too much of a preable.  I wonder if we could create a single shared procedure that takes the patch level as an argument, and then each patch file could contain just a single line like `CALL assert_patch_level(12);`